colcon_cmake/task/cmake/build.py

```diff
@@ -148,25 +148,22 @@ async def _reconfigure(self, args, env):
         cmake_args += ['-DCMAKE_INSTALL_PREFIX=' + args.install_base]
         generator = get_generator(args.build_base, args.cmake_args)
         if os.name == 'nt' and generator is None:
-            vsv = get_visual_studio_version()
-            if vsv is None:
+            try:
+                msbuild_version = get_msbuild_version()
+            except FileNotFoundError as e:
                 raise RuntimeError(
-                    'VisualStudioVersion is not set, '
-                    'please run within a Visual Studio Command Prompt.')
-            supported_vsv = {
-                '16.0': 'Visual Studio 16 2019',
-                '15.0': 'Visual Studio 15 2017',
-                '14.0': 'Visual Studio 14 2015',
-            }
-            if vsv not in supported_vsv:
+                    'msbuild is not found, '
+                    'please run within a Visual Studio Command Prompt.') from e
+            major_version = msbuild_version.split('.')[0]
+            if major_version not in ('14', '15', '16'):
```

Comment:
With the mapping `supported_vsv` removed I don't see a reason to whitelist explicit versions here. Just checking the lower bound (`14`) should be enough.

Reply:
Totally agree. I'm not sure why the upper bound was in there in the first place, but I was keeping existing behavior.

Reply:
> I'm not sure why the upper bound was in there in the first place, but I was keeping existing behavior.

Because the script passed the value of the `supported_vsv` dict to CMake and that value couldn't be computed by the input. You partially avoid the issue by only passing a substring of `-G` and hope that it is unique.